lib/tipg-api/index.ts

```diff
@@ -0,0 +1,112 @@
+import {
+    Stack,
+    aws_ec2 as ec2,
+    aws_rds as rds,
+    aws_lambda as lambda,
+    aws_secretsmanager as secretsmanager,
+    CfnOutput,
+    Duration,
+  } from "aws-cdk-lib";
+  import {
+    PythonFunction,
+    PythonFunctionProps,
+  } from "@aws-cdk/aws-lambda-python-alpha";
+  import { HttpApi } from "@aws-cdk/aws-apigatewayv2-alpha";
+  import { HttpLambdaIntegration } from "@aws-cdk/aws-apigatewayv2-integrations-alpha";
+  import { Construct } from "constructs";
+
+  export class TiPgApiLambda extends Construct {
+    readonly url: string;
+    public tiPgLambdaFunction: PythonFunction;
+
+    constructor(scope: Construct, id: string, props: TiPgApiLambdaProps) {
+      super(scope, id);
+
+      const apiCode = props.apiCode || {
+        entry: `${__dirname}/runtime`,
+        index: "src/handler.py",
+        handler: "handler",
+      };
+
+      this.tiPgLambdaFunction = new PythonFunction(this, "tipg-api", {
+        ...apiCode,
+        runtime: lambda.Runtime.PYTHON_3_10,
+        architecture: lambda.Architecture.X86_64,
+        environment: {
+          PGSTAC_SECRET_ARN: props.dbSecret.secretArn,
+          DB_MIN_CONN_SIZE: "1",
+          DB_MAX_CONN_SIZE: "1",
+          ...props.apiEnv,
+        },
+        vpc: props.vpc,
+        vpcSubnets: props.subnetSelection,
+        allowPublicSubnet: true,
+        memorySize: 1024,
```

Comment:
IMO this should be configurable

Reply:
It makes sense to me as well @vincentsarago.

Should we also make [the titiler-pgstac one configurable ?](https://github.com/developmentseed/eoapi-cdk/blob/daec1e01c4d57f9806b5e34b4d1424fc56146106/lib/titiler-pgstac-api/index.ts#L51) then ?

Reply:
yes, we can make all of them configurable in another PR!